packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md

```diff
@@ -0,0 +1,177 @@
+
+  <Button
+    android:id="@+id/logout_btn"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:layout_marginTop="16dp"
+    android:layout_marginEnd="8dp"
+    android:text="@string/sign_out_btn"
+    android:backgroundTint="@android:color/holo_red_light"
+    android:visibility="gone"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintHorizontal_bias="0.0"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/have_account" />
+
+  <TextView
+    android:id="@+id/have_account"
+    android:layout_width="0dp"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:layout_marginTop="32dp"
+    android:layout_marginEnd="8dp"
+    android:text="@string/have_account"
+    android:textAlignment="center"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintHorizontal_bias="0.53"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"
+    tools:text="@string/have_account" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+3. Fill in strings for the `text` values referenced above, in `app/src/main/res/values/strings.xml`:
+
+```xml
+<resources>
+    <string name="app_name">Browser sign in</string>
+    <string name="sign_in_btn">Sign in</string>
+    <string name="sign_out_btn">Sign out</string>
+    <string name="have_account">Have an account?</string>
+    <string name="welcome_user">Welcome </string>
+</resources>
+```
+
+4. Update the contents of `app/src/main/java/com/okta/android/samples/browser_sign_in/service/AuthClient.java` to initialize the client:
+
+```java
+public class AuthClient {
+  private final static String FIRE_FOX = "org.mozilla.firefox";
+  private final static String CHROME_BROWSER = "com.android.chrome";
+
+  private static AuthClient INSTANCE = null;
+  private static volatile WebAuthClient webAuthClient;
+
+  private AuthClient(Context context) {
+    var config = new OIDCConfig.Builder()
+      .withJsonFile(context, R.raw.okta_oidc_config)
+      .create();
+
+    webAuthClient = new Okta.WebAuthBuilder()
+      .withConfig(config)
+      .withContext(context)
+      .withStorage(new SharedPreferenceStorage(context))
+      .withCallbackExecutor(Executors.newSingleThreadExecutor())
+      .withTabColor(Color.BLUE)
+      .supportedBrowsers(CHROME_BROWSER, FIRE_FOX)
+      //.setRequireHardwareBackedKeyStore(false) // required for emulators
+      .create();
+    }
+
+  public static WebAuthClient getWebAuthClient(Context context) {
+    if (INSTANCE == null) {
+      INSTANCE = new AuthClient(context);
+    }
+    return webAuthClient;
+  }
+}
+```
+
+5. Create a new file called `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.java` to handle client callbacks and button clicks.
+
+6. Give the class the following content:
```

Comment:
something like below once above step is removed
> Add the following code to your activity, for example, `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.java`, to to handle client callbacks and button clicks:

Reply:
updated